src/components/ec-icon/ec-icon.vue

```diff
@@ -12,27 +11,34 @@
 <script>
 export default {
   name: 'EcIcon',
-  props: {
-    name: {
-      type: String,
-      required: true,
-    },
-    size: {
-      type: Number,
-    },
-    type: {
-      type: String,
-      validator(value) {
-        return ['error', 'info', 'success', 'warning', 'interactive'].includes(value);
-      },
-    },
+  compatConfig: {
+    MODE: 3,
```

Comment:
Those are hints for @vue/compat how to handle the component. If a component has been refactored for Vue 3, we set the MODE to 3 to tell the @vue/compat library to leave this component as it is because it can run in Vue 3 on it's own. If the MODE is 2, it indicates the component was written for Vue 2 and @vue/compat will automatically do lots of patches when an instance of this component is being created so it can run in Vue 3. It also recreates some old/obsolete syntax like `$listeners` or filters so if our old code relies on it, it continues working.

Here's the doc of @vue/compat: https://www.npmjs.com/package/@vue/compat#compat-configuration
You can check there what this lib can patch and what had to be refactored.

Reply:
Compatibility with Vue 3 😀